test/controller.test.ts

```diff
@@ -3250,20 +3250,300 @@ contract('Controller', ([owner, accountOwner1, accountOwner2, accountOperator1,
+      const vaultCounter = new BigNumber(await controllerProxy.getAccountVaultCounter(accountOwner1))
+      const collateralToDeposit = new BigNumber(await shortOtoken.strikePrice()).dividedBy(1e18)
+      const actionArgs = [
+        {
+          actionType: ActionType.MintShortOption,
+          owner: accountOwner1,
+          sender: accountOwner1,
+          asset: shortOtoken.address,
+          vaultId: vaultCounter.toNumber(),
+          amount: '1',
+          index: '0',
+          data: ZERO_ADDR,
+        },
+        {
+          actionType: ActionType.DepositCollateral,
+          owner: accountOwner1,
+          sender: accountOwner1,
+          asset: usdc.address,
+          vaultId: vaultCounter.toNumber(),
+          amount: collateralToDeposit.toNumber(),
+          index: '0',
+          data: ZERO_ADDR,
+        },
+      ]
+      await usdc.approve(marginPool.address, collateralToDeposit, {from: accountOwner1})
+      await expectRevert(controllerProxy.operate(actionArgs, {from: accountOwner1}), 'Controller: system is paused')
+    })
+
+    it('should revert withdrawing collateral when system is paused', async () => {
+      const vaultCounter = new BigNumber(await controllerProxy.getAccountVaultCounter(accountOwner1))
+      const collateralToWithdraw = new BigNumber(await shortOtoken.strikePrice()).dividedBy(1e18)
+      const actionArgs = [
+        {
+          actionType: ActionType.WithdrawCollateral,
+          owner: accountOwner1,
+          sender: accountOwner1,
+          asset: usdc.address,
+          vaultId: vaultCounter.toNumber(),
+          amount: collateralToWithdraw.toNumber(),
+          index: '0',
+          data: ZERO_ADDR,
+        },
+      ]
+      await expectRevert(controllerProxy.operate(actionArgs, {from: accountOwner1}), 'Controller: system is paused')
+    })
+
+    it('should revert burning short otoken when system is paused', async () => {
+      const vaultCounter = new BigNumber(await controllerProxy.getAccountVaultCounter(accountOwner1))
+      const actionArgs = [
+        {
+          actionType: ActionType.BurnShortOption,
+          owner: accountOwner1,
+          sender: accountOwner1,
+          asset: shortOtoken.address,
+          vaultId: vaultCounter.toNumber(),
+          amount: '1',
+          index: '0',
+          data: ZERO_ADDR,
+        },
+      ]
+
+      await expectRevert(controllerProxy.operate(actionArgs, {from: accountOwner1}), 'Controller: system is paused')
+    })
+
+    it('should settle vault when system is paused', async () => {
+      // past time after expiry
+      await time.increase(60 * 61) // increase time with one hour in seconds
+      // set price in Oracle Mock, 150$ at expiry, expire ITM
+      await oracle.setExpiryPrice(
+        await shortOtoken.underlyingAsset(),
+        new BigNumber(await shortOtoken.expiryTimestamp()),
+        new BigNumber(150).times(new BigNumber(10).exponentiatedBy(18)),
+      )
```

Comment:
shouldn't there be a separate function for this?

Reply:
Why comine this and redeemForEmergency?

Reply:
What happens when the otoken has not expired? How will a seller / buyer be able to get their share of $?